lib/declare_schema/model/index_definition.rb

```diff
@@ -33,12 +33,15 @@ def initialize(model, fields, **options)
 
       class << self
         # extract IndexSpecs from an existing table
-        # always includes the PRIMARY KEY index
+        # always includes the PRIMARY KEY index for Model unless it is a HABTM Model.
         def for_model(model, old_table_name = nil)
           t = old_table_name || model.table_name
 
-          primary_key_columns = Array(model.connection.primary_key(t)).presence or
-            raise "could not find primary key for table #{t} in #{model.connection.columns(t).inspect}"
+          habtm_model = model.is_a?(DeclareSchema::Model::HabtmModelShim)
+
+          primary_key_columns = Array(model.connection.primary_key(t)).presence
+            primary_key_columns || habtm_model or
+              raise "could not find primary key for table #{t} in #{model.connection.columns(t).inspect}"
```

Comment:
Looks like there's an extra indentation here that we should remove
```suggestion
          primary_key_columns = Array(model.connection.primary_key(t)).presence
          primary_key_columns || habtm_model or
            raise "could not find primary key for table #{t} in #{model.connection.columns(t).inspect}"
```

Reply:
💢👾

Reply:
fixed in 06cd051076355905fb5f9910c463c2c863fe303c